src/App/component/DashBoardComponent/CodeBaseViews.jsx

```diff
@@ -0,0 +1,191 @@
+import { makeStyles } from '@material-ui/core/styles';
+import { useEffect, useState } from 'react'
+import Axios from 'axios'
+import moment from 'moment'
+import { connect } from 'react-redux'
+
+const useStyles = makeStyles(() => ({
+  totalJobViewsGrid: {
+    listStyle: 'none',
+    display: 'grid',
+    gridTemplateColumns: '1fr 1fr',
+    gridTemplateRows: '1fr 1fr',
+    gap: '2rem',
+    padding: 0,
+  },
+  jobViewsBlock: {
+    flexDirection: 'column',
+    display: 'flex',
+    justifyContent: 'center',
+    alignItems: 'center',
+  },
+  jobTitle: {
+    marginBottom: '1rem',
+    fontWeight: 500,
+    display: 'block',
+  },
+  jobViewsContainer: {
+    border: '1px solid #BB3D00',
+    backgroundColor: '#FFDAC8',
+    color: '#BB3D00',
+    display: 'flex',
+    justifyContent: 'center',
+    alignItems: 'center',
+    padding: '1rem',
+    borderRadius: '50%',
+    width: '6rem',
+    height: '6rem',
+  },
+  jobViews: {
+    fontSize: '3rem',
+    fontWeight: 600,
+  },
+}))
+
+function CodeBaseViews(prop) {
+  const classes = useStyles()
+  const [jobs, setJobs] = useState([])
+  const [currentProject, setCurrentProject] = useState({})
+  const [commitListData, setCommitListData] = useState([])
+  const projectId = localStorage.getItem("projectId")
+  const jwtToken = localStorage.getItem("jwtToken")
+
+  useEffect(() => {
+    const fetchCurrentProject = async () => {
+      try {
+        const response = await Axios.get(`http://localhost:9100/pvs-api/project/1/${projectId}`,
+          { headers: { "Authorization": `${jwtToken}` } })
+        setCurrentProject(response.data)
+      } catch (e) {
+        alert(e.response?.status)
+        console.error(e)
+      }
+    }
+    fetchCurrentProject()
+  }, [])
+
+  const getCommitFromGitHub = async () => {
+    const githubRepo = currentProject.repositoryDTOList.find(repo => repo.type === 'github')
+    if (githubRepo !== undefined) {
+      const query = githubRepo.url.split("github.com/")[1]
+      try {
+        await Axios.post(`http://localhost:9100/pvs-api/github/commits/${query}`, "",
+          { headers: { "Authorization": `${jwtToken}` } })
```

Comment:
In this case, I would suggest merge this API into `getGitHubCommitFromDB`, and change the name of this action to `getGitHubCommit`.
Because the frontend just needs the github commits, not have to decide "how to get that", this decision should be chosen by our backend.

Maybe we can address this changes after code freeze.